corehq/apps/reports/standard/deployments.py

```diff
@@ -243,9 +288,26 @@ def total_records(self):
 
     @property
     def rows(self):
-        users = self.user_query().run()
-        self._total_records = users.total
-        return self.process_rows(users.hits)
+        if self.warehouse:
+            mobile_user_and_group_slugs = set(
+                # Cater for old ReportConfigs
+                self.request.GET.getlist('location_restricted_mobile_worker') +
+                self.request.GET.getlist(ExpandedMobileWorkerFilter.slug)
+            )
+            users = ExpandedMobileWorkerFilter.user_es_query(
+                self.domain,
+                mobile_user_and_group_slugs,
```

Comment:
I don't know much about `user_es_query()`, so just checking: is this vulnerable to code/query injection (since these values came directly from the request)?

Reply:
That is an interesting question. I have not looked too deeply into the implementation of that, or the underlying `ESQuery` functions. It is used like this throughout the codebase though, so if it turns out to be an issue, we will want to start flagging the other instances. It does appear the `user_es_query` parses the the values before sticking them into the ES query, so it looks as though it may be resistant to that but I cannot be certain.
Perhaps @esoergel can weigh in on the resistance as he worked on both that query and the underlying ES library we are using.

Reply:
`user_es_query()` is meant to accept these params from the URL, so it should be safe.  Users could manually enter IDs and the like, but there's a domain check and a location-accessibility check for restricted users.  It doesn't do any querystring type searches.